js/modules/privacy.js

```diff
@@ -0,0 +1,53 @@
+/* eslint-env node */
+
+const Path = require('path');
+
+const isString = require('lodash/isString');
+const compose = require('lodash/fp/compose');
+
+
+const PHONE_NUMBER_PATTERN = /\+\d{7,12}(\d{3})/g;
+const GROUP_ID_PATTERN = /(group\()([^)]+)(\))/g;
+
+const APP_ROOT_PATH = Path.join(__dirname, '..', '..', '..');
+const APP_ROOT_PATH_PATTERN = new RegExp(APP_ROOT_PATH, 'g');
```

Comment:
We're creating a regular expression from an arbitrary incoming path. Seems like we may want to put some protections in here to prevent crashes. Or perhaps rely on a recursive string replace instead of using regular expressions.

Also, doesn't the `path.join()` code above create a path ending with `../../..`? `path.resolve()` should help with that.

Reply:
Re: `RegExp`. Good call on being more defensive around creating the `RegExp` from a user supplied file path. Followed JDD’s advice and added `_.escapeRegExp`: https://github.com/lodash/lodash/issues/1084#issuecomment-86698786. Wish there was simply a `String::replaceAll` function.

Re: `path.join`. Confirmed it does the right thing and normalizes the path to not have `../../..`. Otherwise, the tests wouldn’t have passed anyways. I prefer `path.join` to `path.resolve` if it meets our needs as it is a pure function. `path.resolve` falls back `process.cwd()`.

Reply:
FWIW, PureScript offers a straightforward `String::replaceAll` function: https://pursuit.purescript.org/packages/purescript-strings/3.5.0/docs/Data.String#v:replaceAll

The implementation shows that it’s non-trivial: https://github.com/purescript/purescript-strings/blob/06e048c1a6026e3fd6dd9aff0a2d26da23cec783/src/Data/String.js#L114-L120

😜